p8e-util/src/main/kotlin/io/p8e/crypto/SmartKeySigner.kt

```diff
@@ -65,14 +60,19 @@ class SmartKeySigner(
         signatureRequest = SignRequest()
             .hashAlg(DigestAlgorithm.SHA512)
             .deterministicSignature(true)
-            .data(byteArrayOf())
+            .hash(byteArrayOf())
     }
 
     override fun update(data: ByteArray, off: Int, res: Int) {
-        signatureRequest?.data = data.copyOfRange(off, res)
+        messageDigest.update(data, off, res)
     }
 
-    override fun sign(): ByteArray = signAndVerifyApi.sign(keyUuid, signatureRequest).signature
+    override fun sign(): ByteArray {
+        // Completes the message digest and adds the necessary padding before signing
+        signatureRequest?.hash = messageDigest.digest()
```

Comment:
before the standup i had this in the `update` method, so this caused the padding to be off🤦

Reply:
Less bytes over the network as well :)